packages/devtools/test/viewport_canvas_test.dart

```diff
@@ -17,6 +18,10 @@ Future<void> settleUi() async {
 }
 
 void main() {
+  // Use a floating point device pixel ratio to put more pressure on the canvas
+  // logic to round pixel values correctly.
+  environment.overrideDevicePixelRatio(2.3);
```

Comment:
👍; this reminds me of CIs that run all the tests from a directory that contains spaces in the path, just so those cases can't regress

Reply:
yep! fyi you can also run the devtools app setting the pixel ratio in main() to see what crazy DPIs look like.